File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/command/builder/CommandBuilderArg.java
package pize.tests.voxelgame.main.command.builder;

import pize.tests.voxelgame.main.command.Command;
import pize.tests.voxelgame.main.command.argument.CommandArg;
import pize.tests.voxelgame.main.command.node.CommandNodeArg;

public class CommandBuilderArg{
    
    private final CommandNodeArg commandRoot;
    
    public CommandBuilderArg(String name, CommandArg argument){
        commandRoot = new CommandNodeArg(name, argument);
    }
    
    
    public CommandBuilderArg then(String literal, CommandArg argument){
        commandRoot.addChild(new CommandNodeArg(literal, argument));
        return this;
    }
    
    public CommandBuilderArg executes(Command command){
        return this;
    }
    
    
    public CommandNodeArg buildArg(){
        return commandRoot;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/command/builder/CommandBuilderArg.kt
package pize.tests.voxelgame.main.command.builder

import pize.tests.voxelgame.main.command.Command
import pize.tests.voxelgame.main.command.argument.CommandArg
import pize.tests.voxelgame.main.command.node.CommandNodeArg

class CommandBuilderArg(name: String, argument: CommandArg) {
    private val commandRoot: CommandNodeArg

    init {
        commandRoot = CommandNodeArg(name, argument)
    }

    fun then(literal: String, argument: CommandArg): CommandBuilderArg {
        commandRoot.addChild(CommandNodeArg(literal, argument))
        return this
    }

    fun executes(command: Command?): CommandBuilderArg {
        return this
    }

    fun buildArg(): CommandNodeArg {
        return commandRoot
    }
}
